Project Mars/MarsQA1/MarsQA-1/Specflow pages/Pages/Language_page.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Threading;
+using Gherkin.CucumberMessages.Types;
+using MarsQA1.Helper;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using TechTalk.SpecFlow;
+
+namespace MarsQA1.Specflow_Pages;
+
+public class Language_page
+{
+    
+    public void LanguageTab(IWebDriver driver)
+    {
+        Wait.WaitToBeClickable(driver, "XPath","//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/thead/tr/th[3]/div" , 3); 
+        Driver.driver.FindElement(By.XPath(
+            "//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/thead/tr/th[3]/div")).Click();
+    }
+
+    public void AddNew(IWebDriver driver,string language, string dropdownValue)
+    {
+        //Enter the language in the TextBox
+        Driver.driver.FindElement(By.Name("name")).SendKeys(language);
+        
+       //Find the Dropdown Element
+        Wait.WaitToBeClickable(driver, "Name", "level", 5);
+        IWebElement dropdown = Driver.driver.FindElement(By.Name("level"));
+        SelectElement select = new SelectElement(dropdown);
+        select.SelectByText(dropdownValue);
+        
+    }
+
+    public void ClickAddButton()
+    {
+        Driver.driver.FindElement(By.XPath("//*[@id=\'account-profile-section\']/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/div/div[3]/input[1]")).Click();
+    }
+
+    public void Add_Verify(IWebDriver driver,string validationMessage)
+    {
+       try
+       {
+           Wait.WaitToBeVisible(driver, "XPath", "//div[@class='ns-box-inner']", 5);
+           IWebElement toastElement = Driver.driver.FindElement(By.XPath("//div[@class='ns-box-inner']"));
+
+           var scenarioTitle = ScenarioContext.Current.ScenarioInfo.Title;
+           switch (scenarioTitle)
+           {
+               case "Scenario1":
+                   Assert.AreEqual("English has been added to your languages",toastElement.Text);
+                   break;
+               case "Scenario2":
+                   Assert.AreEqual("Please enter language and level",toastElement.Text);
+                   break;
+               case "Scenario3":
+                   Assert.AreEqual("Please enter language and level",toastElement.Text);
+                   break;
+               case "Scenario4":
+                   Assert.AreEqual("This language is already exist in your language list",toastElement.Text);
```

Comment:
also this is not correct you should not add switch statement as it gives you false positive results you can test this when you add the add test twice, you will see on the second time it will show language already exists which is wrong as your test should test adding a language not adding a language that already exists. What you have not done is thought about state  for example for the add test you need to first delete all languages present in the table and then add language, and if you were updating a language you will first need to reset state (no languages presents in the table) and then add then update please think about state for all the tests language and skills both